modules/concurrency/src/spinner.cpp

```diff
@@ -39,7 +39,7 @@ Spinner::Spinner(Callback&& callback, double rate_hz /*= 0*/)
 }
 
 Spinner::~Spinner() {
-  if (is_started_.load() || spinner_thread_.joinable()) {
+  if (is_started_.load()) {
```

Comment:
we can remove `is_started_` and replace this with:
```suggestion
  if (async_spinner_handle_.valid()) {
```

Reply:
This would work, if we would not move `async_spinner_handle_` in `stop()`. Since we do, that change would seg fault. We need the helper variable `is_started_` since we cannot know if `async_spinner_handle_` has been moved.

Reply:
actually, it is legal to call `valid` on moved object according to documentation https://en.cppreference.com/w/cpp/thread/future/valid